contracts/src/system/README

```diff
@@ -0,0 +1,19 @@
+# Public Callbacks Contract
+
+The PublicCallbacks contract provides a mechanism for registering and executing callbacks in a gas-efficient manner. It is designed to be used by contracts that need to hide their gas cost whenever executing something and make the state change immutable if proxied.
```

Comment:
not only hide gas cost. Also make the action impossible to revert if the result is not positive